workloads/pvcscale.yml

```diff
@@ -0,0 +1,144 @@
+      failed_when: scale_ci_tooling_project.rc == 0
+      until: scale_ci_tooling_project.rc == 1
+      retries: 60
+      delay: 1
+      when: scale_ci_tooling_ns_exists.rc == 0
+
+    - name: Block for non-existing tooling namespace
+      block:
+        - name: Create tooling namespace
+          shell: |
+            oc create -f {{ansible_user_dir}}/scale-ci-tooling/scale-ci-tooling-ns.yml
+
+        - name: Create tooling service account
+          shell: |
+            oc create serviceaccount useroot -n scale-ci-tooling
+            oc adm policy add-scc-to-user privileged -z useroot -n scale-ci-tooling
+          when: enable_pbench_agents|bool
+      when: scale_ci_tooling_ns_exists.rc != 0
+
+    - name: PVC test specific user configuration
+      block:
+        - name: Add htpasswd identity provider for user when running pvc scale test
+          shell: |
+            htpasswd -c -B -b /tmp/htpasswd useroot useroot
+            oc delete secret -n openshift-config htpass-secret
+            oc create secret generic htpass-secret --from-file=/tmp/htpasswd -n openshift-config
+
+    - name: Create/replace kubeconfig secret
+      shell: |
+        oc replace --force -n scale-ci-tooling -f "{{ansible_user_dir}}/scale-ci-tooling/kubeconfig-secret.yml"
+
+    - name: Create/replace the pbench configmap
+      shell: |
+        oc replace --force -n scale-ci-tooling -f "{{ansible_user_dir}}/scale-ci-tooling/pbench-cm.yml"
+
+    - name: Create/replace pbench ssh secret
+      shell: |
+        oc replace --force -n scale-ci-tooling -f "{{ansible_user_dir}}/scale-ci-tooling/pbench-ssh-secret.yml"
+
+    - name: Create/replace workload script configmap
+      shell: |
+        oc replace --force -n scale-ci-tooling -f "{{ansible_user_dir}}/scale-ci-tooling/workload-pvcscale-script-cm.yml"
+
+    - name: Create/replace workload job to that runs workload script
+      shell: |
+        oc replace --force -n scale-ci-tooling -f "{{ansible_user_dir}}/scale-ci-tooling/workload-job.yml"
+
+    - name: Poll until job is active
+      shell: |
+        oc get job scale-ci-pvcscale -n scale-ci-tooling -o json
+      register: job_json
+      retries: 60
+      delay: 2
+      until: job_json.stdout | from_json | json_query('status.active==`1`')
+
+    - name: Poll until job is complete
+      shell: |
+        oc get job scale-ci-pvcscale -n scale-ci-tooling -o json
+      register: job_json
+      retries: "{{job_completion_poll_attempts}}"
+      delay: 10
+      until: job_json.stdout | from_json | json_query('status.succeeded==`1` || status.failed==`1`')
+      failed_when: job_json.stdout | from_json | json_query('status.succeeded==`1`') == false
+      when: job_completion_poll_attempts|int > 0
+
+    - name: remove user secret for user useroot - we needed it only for pvc scale test
+      shell: |
+        oc delete secret htpass-secret -n openshift-config
+
+    - name: delete PVC project after test is done
```

Comment:
I am not sure this is a good idea in that an"unspoken" feature is to set the `job_completion_poll_attempts` to 0 so the playbook just kicks off the job and then in CI we can `oc logs -f` the workload container for workload status.  I'd rather see this be a parameter-ized feature "in-band" of the script with an example being like the network tests.  See this Example: https://github.com/openshift-scale/workloads/blob/master/workloads/files/workload-network-script-cm.yml#L104-L123

This permits us to debug the script (Set cleanup to false) and retain the CI ability to tail the logs of the workload container.

Reply:
Ok, it can be paramter  to satisfy CI part, I will update

Reply:
Got it thanks, my apoligizes about the the "unspoken" ci requirements.  The other reason to do it inband of the workload script is such that CI can still run it and it will still cleanup when it is done.

Reply:
for this test, if someone set `cleanup: true` then it is going to clean up project before data are collected, so I had to make it impossible to clean up project with pods/pvcs and delete it at end of test when it is not any more necessary. `cleanup: true` is part of clusterloader and in first step it creates objects and if `cleanup: true` it cleans them at end of `clusterloader` part, timestamp collection is outside of clusterloader section.

Reply:
Got it, so we should allow cleanup, but after the post processing component.  I imagine it would be around [line 55](https://github.com/openshift-scale/workloads/pull/24/files#diff-0039e9acca516519e319ae6e828a4020R55).

Reply:
yes, I will test that scenario and update PR.